eWAYRecurring.php

```diff
@@ -436,11 +437,12 @@ function validateEwayContribution($paymentProcessor, $invoiceID) {
       require_once extensionPath('vendor/autoload.php');
 
       $contribution = $contribution['values'][0];
-      $eWayAccessCode = CRM_Utils_Request::retrieve('AccessCode', 'String', $form, FALSE, "");
-      $qfKey = CRM_Utils_Request::retrieve('qfKey', 'String', $form, FALSE, "");
-
-      $paymentProcessor->validateContribution($eWayAccessCode, $contribution, $qfKey, $paymentProcessor->getPaymentProcessor());
+      if ($contribution['total_amount'] != '0.00') {
+        $eWayAccessCode = CRM_Utils_Request::retrieve('AccessCode', 'String', $form, FALSE, "");
+        $qfKey = CRM_Utils_Request::retrieve('qfKey', 'String', $form, FALSE, "");
 
+        $paymentProcessor->validateContribution($eWayAccessCode, $contribution, $qfKey, $paymentProcessor->getPaymentProcessor());
+      }
```

Comment:
Thanks @agileware-justin I think changes in this file is still required since the new contribution created is validated in the extension even if total amount is 0?